core/server/master/src/main/java/alluxio/master/throttle/ServerIndicator.java

```diff
@@ -0,0 +1,270 @@
+   *
+   * @param directMemUsed the used direct mem
+   * @param ratioOfUsedHeap the ratio of used heap
+   * @param cpuLoad the cpu load
+   * @param totalJVMPauseTimeMS the total JVM pause time
+   * @param rpcQueueSize the rpc queue size
+   * @return  the threshold indicator object
+   */
+  public static ServerIndicator createThresholdIndicator(long directMemUsed,
+      double ratioOfUsedHeap, double cpuLoad, long totalJVMPauseTimeMS, long rpcQueueSize) {
+    long pitTotalJVMPauseTimeMS = 0;
+    long heapMax = (long) (MetricsSystem.METRIC_REGISTRY
+        .gauge(MetricsMonitorUtils.MemoryGaugeName.HEAP_MAX, null).getValue());
+    long heapUsed = (long) (ratioOfUsedHeap * heapMax);
+
+    if (Configuration.getBoolean(PropertyKey.MASTER_JVM_MONITOR_ENABLED)) {
+      pitTotalJVMPauseTimeMS = (long) (MetricsSystem.METRIC_REGISTRY
+          .gauge(MetricsMonitorUtils.ServerGaugeName.TOTAL_EXTRA_TIME, null).getValue());
+    }
+    return new ServerIndicator(directMemUsed, heapMax,
+        heapUsed, cpuLoad, totalJVMPauseTimeMS, pitTotalJVMPauseTimeMS,
+        rpcQueueSize, System.currentTimeMillis());
+  }
+
+  /**
+   * @return the direct mem used
+   */
+  public long getDirectMemUsed() {
+    return mDirectMemUsed;
+  }
+
+  /**
+   * @return the max heap
+   */
+  public long getHeapMax() {
+    return mHeapMax;
+  }
+
+  /**
+   * @return the used heap
+   */
+  public long getHeapUsed() {
+    return mHeapUsed;
+  }
+
+  /**
+   * @return the cpu load
+   */
+  public double getCpuLoad() {
+    return mCpuLoad;
+  }
+
+  /**
+   * @return the total JVM pause time
+   */
+  public long getTotalJVMPauseTimeMS() {
+    return mTotalJVMPauseTimeMS;
+  }
+
+  /**
+   * @return the PIT total JVM pause time
+   */
+  public long getPITTotalJVMPauseTimeMS() {
+    return mPITTotalJVMPauseTimeMS;
+  }
+
+  /**
+   * @return the rpc queue size
+   */
+  public long getRpcQueueSize() {
+    return mRpcQueueSize;
+  }
+
+  /**
+   * @return the time to measure the indicators
+   */
+  public long getPitTimeMS() {
+    return mPitTimeMS;
+  }
+
+  /**
+   * Sets the pit time.
+   *
+   * @param pitTimeMS the pit time
+   */
+  public void setPitTimeMS(long pitTimeMS) {
+    mPitTimeMS = pitTimeMS;
+  }
+
+  /**
+   * Addition the server indicator object.
+   *
+   * @param serverIndicator the server indicator
+   */
+  public void addition(ServerIndicator serverIndicator) {
+    // This is to record the aggregated time
+    mDirectMemUsed += serverIndicator.getDirectMemUsed();
+    mRpcQueueSize += serverIndicator.getRpcQueueSize();
+    mCpuLoad += serverIndicator.getCpuLoad();
+    mHeapUsed += serverIndicator.getHeapUsed();
+    mTotalJVMPauseTimeMS += serverIndicator.getTotalJVMPauseTimeMS();
+  }
+
+  /**
+   * Gets the delta the server indicator based on server indicator.
+   *
+   * @param serverIndicator the base server indicator
+   */
+  public void reduction(ServerIndicator serverIndicator) {
```

Comment:
You can see non of them should be negative

Reply:
If negative value found, it should be a bug, need to be fixed.
Here just make sure it is not impacting on the real logic.

Reply:
Maybe delta is the wrong description to use in the comment then? Also maybe I don't understand what these methods are being used for so maybe its worth describing that? Like is the addition being used so that an average can be computed later?

Reply:
For example:
aggregate 1 = (i1 + i2 +i3)
the reduction would be used to calculate 
aggregate 2 =  i2 + i3 + i4,
I don't want to calculate that much every time so 
aggregate 2 = aggregate 1 - i1 + i4

Reply:
The reduction would do the following:
(this - serverIndicator) 

Which term is more comfortable for you?

Reply:
Yes thanks, this makes more sense now with the comment.